src/Data/ProgrammesDb/EntityRepository/VersionRepository.php

```diff
@@ -209,4 +222,86 @@ public function createQueryBuilder($alias, $indexBy = null)
             ->addSelect('p')
             ->join($alias . '.programmeItem', 'p');
     }
+
+    /**
+     * This deals with the case where we want to query competition warnings
+     * for a version where...
+     * a) the version's programmeItem is not directly linked to a masterbrand, but
+     * b) the programmeItem's ancestor is directly linked to a masterbrand.
+     *
+     * It ensures those things get the correct competition warning.
+     *
+     * In 90% of cases, this will just return the input array.
+     *
+     * In the other cases it queries all the programme's parents, and their masterbrands
+     * joined to their competition warnings, adds them to the version, and returns that
+     * whole mess.
+     *
+     * @param array $versions
```

Comment:
I think this also make sense
`@param version[] $versions`

Reply:
It's not though, it's the raw array that comes back from Doctrine, not an array of Version objects